crates/ir/src/fact.rs

```diff
@@ -357,6 +359,19 @@ impl PropIdx {
             }
         }
     }
+
+    pub fn valid(
+        &self,
+        ctx: &(impl Ctx<Prop>
```

Comment:
Use a context bound: `valid<Ctx>(..) where Ctx: ...`

Reply:
This doesn't match with the existing functions `relevant_vars` etc though. Should I change all of them or just valid?

```
pub fn relevant_vars(
        &self,
        ctx: &(impl Ctx<Time> + Ctx<Prop> + Ctx<Expr>),
 ) -> ...
 ```